### DISCUSSION TITLE
"Noisy" wannabe input for dragon resource tracker

### DISCUSSION THREAD

**Discussion by mpourmpoulis** [dictation-toolbox/Caster#825] [Show and tell]
Ok, as I have posted about these Quite some time ago in the chat ,Inspired by [Talon](https://noise.talonvoice.com/) I have been playing around with using sounds with Dragon. Now the results are clearly inferior compared to what a dedicated noise recognition engine can achieve, but on my end DNS 15 Home can recognize some 30+ letter like "sounds" like "fffp,pppf, tttf,shhh,ssss/'s,shhp,pppt,xxxx,tttp,kkkp" , you just have to make sure that you use 4 or more letters in your grammar (so for instance ffp will not work), which is good enough to still make a difference for quite a lot of simple tasks. 

I am opening up this issue to

- Make the existence of This feature visible to more users

- see if others Can reproduce this behavior

- Motivate people to give it a try with the other available backends  or if they already have , to share their results

edit: my microphone is FlexyMike Dual Ear Cardioid(DEC)

**Comment by LexiconCode** [comment]
If you have a grammar that standalone to demonstrate these different sounds that would be the 1st step for others to test.

**Comment by mpourmpoulis** [comment]
Yeah, that's probably a good idea. I have most of my "sounds" Embedded in other grammars, so I quickly scrambled this together

```python
from dragonfly import Repeat, Pause, Function, Choice, MappingRule,Dictation,Mimic

from castervoice.lib.actions import Key, Mouse, Text

from castervoice.lib.merge.additions import IntegerRefST
from castervoice.lib.ctrl.mgr.rule_details import RuleDetails
from castervoice.lib.merge.state.short import R

from castervoice.lib import github_automation
from castervoice.lib.temporary import Store, Retrieve


class NoiseChromeRule(MappingRule):
    mapping = {
        "shhh":R(Key("pagedown")),
        "shhh shhh":R(Key("pagedown:4")),
        "shhh shhh shhh":R(Key("pagedown:8")),
        "long shh":R(Key("pagedown:3")),

        "shhf":R(Key("pageup")),
        "tttf":R(Key("pageup")),
        "pppf":R(Key("pageup")),
        "shhf shhf":R(Key("pageup:4")),

        "'s":R(Key("f"),rdescript = "trigger Vimium"),
        "'s 's":R(Key("F"),rdescript = "trigger Vimium"),


        "shhp":R(Key("c-pagedown"),rdescript = "next tab"),
        "shhp shhp":R(Key("c-pagedown:4")),
        "shhp shhp shhp":R(Key("c-pagedown:8")),

        "pppt":R(Key("c-pageup"),rdescript = "previous tab"),
        "pppt pppt":R(Key("c-pageup:2")),
        "pppt pppt pppt":R(Key("c-pageup:3")),


        "tttt":R(Mimic("new","tab")),
        "ffft":R(Mimic("new","tab")),
        "fffk":R(Key("c-w"),rdescript = "close tab"),

        
        "xxxp":R(Mimic("go","back")),
        "xxxf":R(Mimic("go","forward")),
        
        
    }
    extras = []
    defaults = {}


def get_rule():
    return NoiseChromeRule, RuleDetails(name="noise chrome", executable="chrome")
```

Edit: It should be rdescript instead of rdscript
Edit: Fixed mistake with Key action

**Reply by pokey** [reply]
Cool! Fwiw if you are looking for a dedicated noise engine, you might check out [parrot.py](https://github.com/chaosparrot/parrot.py). Still a bit rough around the edges but I've been using it for the past week with talon and it's been awesome. I'd imagine you could fairly easily combine it with caster as well
